services/src/main/java/org/keycloak/protocol/saml/EntityDescriptorDescriptionConverter.java

```diff
@@ -187,6 +187,7 @@ private static ClientRepresentation loadEntityDescriptors(InputStream is) {
         attributes.put(SamlConfigAttributes.SAML_SIGNATURE_ALGORITHM, SignatureAlgorithm.RSA_SHA256.toString());
         attributes.put(SamlConfigAttributes.SAML_AUTHNSTATEMENT, SamlProtocol.ATTRIBUTE_TRUE_VALUE);
         SPSSODescriptorType spDescriptorType = getSPDescriptor(entity);
+        if (spDescriptorType == null) return app;
```

Comment:
This method is to create a SAML client via the  `SPSSODescriptor`. So, if the descriptor is not for an SP, maybe it's better to throw an exception. Because it makes no sense returning something that will be completely empty or filled with default values. Similar to the case when there is no entity descriptor a few lines before. Something like:

```java
if (spDescriptorType == null) {
    throw new RuntimeException("No SPSSODescriptorType defined in the entity descriptor file");
}
```

If you prefer you can return a `BadRequestException`. Because, just curiosity, what are you trying to import here? A IDPSSODescriptor or another thing?